test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperUsageDescriptorFactoryTest.cs

```diff
@@ -0,0 +1,316 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if !DNXCORE50
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNet.Razor.TagHelpers;
+using Microsoft.AspNet.Testing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class TagHelperUsageDescriptorFactoryTest
+    {
+        // The test assembly doesn't really exist. This is used to look up corresponding XML for the fake assembly
+        // which is based on the DocumentedTagHelper type.
+        public static readonly string DocumentedTestAssemblyLocation =
+            Path.ChangeExtension(DocumentedTagHelper.XmlTestFileLocation, ".dll");
+        public static readonly string DocumentedTestAssemblyCodeBaseLocation =
+            "file:///" + DocumentedTestAssemblyLocation;
+
+        public static TheoryData CreateDescriptor_TypeData
+        {
+            get
+            {
+                var defaultLocation = DocumentedTestAssemblyLocation;
+                var defaultCodeBase = DocumentedTestAssemblyCodeBaseLocation;
+                var nonExistentLocation = defaultLocation.Replace("TestFiles", "TestFile");
+                var nonExistentCodeBase = defaultCodeBase.Replace("TestFiles", "TestFile");
+                var invalidLocation = defaultLocation + '\0';
+                var invalidCodeBase = defaultCodeBase + '\0';
+
+                // tagHelperType, expectedUsageDescriptor
+                return new TheoryData<Type, TagHelperUsageDescriptor>
+                {
+                    { CreateDocumentationTagHelperTestType(location: null, codeBase: null), null },
+                    {
+                        CreateDocumentationTagHelperTestType(defaultLocation, codeBase: null),
+                        new TagHelperUsageDescriptor(
+                            summary: "The summary for the type.",
```

Comment:
`<summary>` and `<remarks>` rarely contain just simple text. Need to test more common content such as `<see cref="T:Microsoft.AspNet.Razor.Runtime.TagHelpers.ITagHelper"/> implementation targeting &lt;a&gt; elements.` and `Must be <c>null</c> if <see cref="P:Microsoft.AspNet.Mvc.TagHelpers.AnchorTagHelper.Route"/> is non-<c>null</c>.`.